src/ServiceControl.AcceptanceTesting/Support/EndpointRunner.cs

```diff
@@ -105,6 +105,10 @@ public class EndpointRunner
                                 {
                                     executedWhens.Add(when.Id);
                                 }
+                                else
```

Comment:
Why are we mixing async code path with Thread.Sleep? Would a Task.Delay also work? If yes why would we then need the explicit yield at the end?

Reply:
Then how about only doing a Task.Delay ?